sentry-core/src/main/java/io/sentry/core/SentryEnvelopeHeaderAdapter.java

```diff
@@ -24,9 +25,44 @@ public void write(JsonWriter writer, SentryEnvelopeHeader value) throws IOExcept
       writer.value(value.getEventId().toString());
     }
 
+    final SdkInfo sdkInfo = value.getSdkInfo();
+    if (sdkInfo != null) {
+      boolean hasInitAttrs = false;
+
+      if (sdkInfo.getSdkName() != null) {
+        hasInitAttrs = initAttrs(writer, hasInitAttrs);
+        writer.name("sdk_name").value(sdkInfo.getSdkName());
+      }
+      if (sdkInfo.getVersionMajor() != null) {
+        hasInitAttrs = initAttrs(writer, hasInitAttrs);
+        writer.name("version_major").value(sdkInfo.getVersionMajor());
+      }
+      if (sdkInfo.getVersionMinor() != null) {
+        hasInitAttrs = initAttrs(writer, hasInitAttrs);
+        writer.name("version_minor").value(sdkInfo.getVersionMinor());
+      }
+      if (sdkInfo.getVersionPatchlevel() != null) {
+        hasInitAttrs = initAttrs(writer, hasInitAttrs);
+        writer.name("version_patchlevel").value(sdkInfo.getVersionPatchlevel());
+      }
```

Comment:
What about we only add the version numbers if we have a name?

```suggestion
      if (sdkInfo.getSdkName() != null) {
        hasInitAttrs = initAttrs(writer, hasInitAttrs);
        writer.name("sdk_name").value(sdkInfo.getSdkName());
        if (sdkInfo.getVersionMajor() != null) {
          writer.name("version_major").value(sdkInfo.getVersionMajor());
        }
        if (sdkInfo.getVersionMinor() != null) {
          writer.name("version_minor").value(sdkInfo.getVersionMinor());
        }
        if (sdkInfo.getVersionPatchlevel() != null) {
          writer.name("version_patchlevel").value(sdkInfo.getVersionPatchlevel());
        }
      }
```

Reply:
makes sense